### PR TITLE
[2.4] Improve reliable message

### PR DESCRIPTION
**Pull request by yanchengnv**
Fixes # .

### Description

This PR improves the ReliableMessage (RM) mechanism:

- Automatically enable RM so that user won't need to manually configure
- No need to set sender_id in XGB Executor configuration
- Added detailed log messages for tracing/debugging RM process

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/app_opt/xgboost/histogram_based_v2/adaptors/grpc_client_adaptor.py [hunk 1]
```diff
@@ -87,21 +87,17 @@ class GrpcClientAdaptor(XGBClientAdaptor, FederatedServicer):
            federated gRPC client.
     """
 
-    def __init__(
-        self,
-        int_server_grpc_options=None,
-        in_process=False,
-        req_timeout=100,
-    ):
+    def __init__(self, int_server_grpc_options=None, in_process=False, req_timeout=10.0, tx_timeout=100.0):
```
Comment (nvidianz) [inline nvflare/app_opt/xgboost/histogram_based_v2/adaptors/grpc_client_adaptor.py:90]: Is req_timeout same as msg_timeout? If so, we should use the same name.

File: nvflare/app_opt/xgboost/histogram_based_v2/adaptors/grpc_server_adaptor.py [hunk 2]
```diff
@@ -132,7 +132,7 @@ def start(self, fl_ctx: FLContext):
         if not port:
             raise RuntimeError("failed to get a port for XGB server")
 
-        server_addr = f"localhost:{port}"
+        server_addr = f"127.0.0.1:{port}"
```
Comment (YuanTingHsieh) [inline nvflare/app_opt/xgboost/histogram_based_v2/adaptors/grpc_server_adaptor.py:135]: @nvidianz @IsaacYangSLA mentioned something about localhost is not on Azure, so this is a good change.

**Review by nvidianz — APPROVED** [review]
Looks good.

File: nvflare/apis/utils/reliable_message.py [hunk 3]
```diff
@@ -93,69 +104,93 @@ def process(self, request: Shareable, fl_ctx: FLContext) -> Shareable:
             # it is possible that a new request for the same tx is received while we are processing the previous one
             if not self.rcv_time:
                 self.rcv_time = time.time()
-                self.timeout = request.get_header(HEADER_TIMEOUT)
+                self.per_msg_timeout = request.get_header(HEADER_PER_MSG_TIMEOUT)
+                self.tx_timeout = request.get_header(HEADER_TX_TIMEOUT)
 
                 # start processing
+                ReliableMessage.info(fl_ctx, f"started processing request of topic {self.topic}")
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply nvflare/apis/utils/reliable_message.py:111]: but let's remove them after we done with all testing

File: nvflare/apis/utils/reliable_message.py [hunk 4]
```diff
@@ -294,110 +438,139 @@ def _send_request(
...
 
         # Querying phase - try to get result
         engine = fl_ctx.get_engine()
         query = Shareable()
-        query.set_header(HEADER_TX, receiver.tx_id)
+        query.set_header(HEADER_TX_ID, receiver.tx_id)
         query.set_header(HEADER_OP, OP_QUERY)
 
         num_tries = 0
+        last_query_time = 0
```
Comment (YuanTingHsieh) [inline nvflare/apis/utils/reliable_message.py:524]: Do we update this last_query_time somewhere?

File: nvflare/apis/utils/reliable_message.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (yanchengnv) [reply nvflare/apis/utils/reliable_message.py:111]: Yes, that's the idea. We want to have a lot of logging when developing new features. We could change to debug level later.

File: nvflare/apis/utils/reliable_message.py (hunk 4 above)
Comment (yanchengnv) [reply nvflare/apis/utils/reliable_message.py:524]: good catch! fixed.

**Review by YuanTingHsieh — APPROVED** [review]

**Pull request merged**